webpack.config.js

```diff
@@ -0,0 +1,179 @@
+const path = require("path");
+const HtmlWebPackPlugin = require("html-webpack-plugin");
+const ReactRefreshWebpackPlugin = require("@pmmmwh/react-refresh-webpack-plugin");
+const TerserPlugin = require("terser-webpack-plugin");
+const CopyPlugin = require("copy-webpack-plugin");
+const ForkTsCheckerWebpackPlugin = require("fork-ts-checker-webpack-plugin");
+const { createBabelConfig } = require('./babel.config.js');
+const { DefinePlugin, ProvidePlugin } = require("webpack");
+const JSONManifestPlugin = require("./jsonmanifest-plugin");
+
+const createStats = verbose => ({
+    assets: verbose,
+    builtAt: verbose,
+    cached: false,
+    children: false,
+    chunks: false,
+    colors: true,
+    entrypoints: true,
+    hash: false,
+    modules: false,
+    performance: false,
+    timings: verbose,
+    version: verbose,
+});
+
+module.exports = (env, argv) => ({
+    context: path.resolve(__dirname),
+    // Our application entry point.
+    entry: {
+        main: [
+            "./src",
+        ],
+    },
+    devtool: 'inline-source-map',
+    module: {
+        rules: [
+            {
+                test: /\.[jt]sx?$/,
+                exclude: /node_modules/,
+                use: [
+                    {
+                        loader: "babel-loader",
+                        options: createBabelConfig({ mode: argv.mode }),
+                    },
+                ],
+            },
+            {
+                test: /\.css$/,
+                exclude: /node_modules/,
+                use: ["style-loader", "css-loader"],
+                sideEffects: true,
+            },
+            {
+                test: /\.(scss)$/,
+                use: ["style-loader", "css-loader", "postcss-loader", "fast-sass-loader"],
+                sideEffects: true,
+            },
+            {
+                test: /\.(png|jpe?g|gif|svg)$/i,
+                loader: 'file-loader',
+            },
```

Comment:
File loader has been deprecated, use asset/resource
```suggestion
            {
                test: /\.(png|jpe?g|gif|svg)$/i,
                loader: 'asset/resource',
                generator: {
                    publicPath: process.env.TGS_WEBPANEL_GITHUB_PATH || (mode === "production" ? "/app/" : "/")
                }
            },
```

Reply:
loader is wrong, its type